src/openloco/window.cpp

```diff
@@ -605,8 +605,8 @@ namespace openloco::ui
 
         auto pos = coordinate_3d_to_2d(loc.x, loc.y, loc.z, WindowManager::getCurrentRotation());
 
-        pos.x -= viewport->view_width * -32768;
-        pos.y -= viewport->view_height * -32768;
+        pos.x -= (std::abs(viewport->view_width * -32768) >> 16);
```

Comment:
just change it to `viewport->view_width / 2` its much easier to understand that way